### PR TITLE
Update to python3.9

### PR DESCRIPTION
**Pull request by j0nnyr0berts**
* Bump to python3.9 and update circleci config
* Unpin dev packages and address type hinting issues

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +18,7 @@ clean:
 
 sync:
 	pipenv sync --dev
+	pipenv run pip install -e .
```
Comment (j0nnyr0berts) [inline Makefile:21]: Pipenv does not install tentaclio from the Pipefile 🙄

File: src/tentaclio/fs/remover.py [hunk 2]
```diff
@@ -10,20 +8,9 @@
 __all__ = ["REMOVER_REGISTRY", "ClientRemover"]
 
 
-class Remover(Protocol):
+class Remover(ContextManager, Protocol):
```
Comment (j0nnyr0berts) [inline src/tentaclio/fs/remover.py:11]: Inheriting from both ContextManager + Protocol seems to work now

File: src/tentaclio/protocols.py [hunk 3]
```diff
@@ -51,13 +49,13 @@ class Writer(_Writer[Any], Protocol):
     ...
 
 
-class ByteReader(_Reader[ByteString], Protocol):
```
Comment (j0nnyr0berts) [inline src/tentaclio/protocols.py:54]: typing.BytesString is [deprecated in 3.9](https://www.python.org/dev/peps/pep-0585/) which
is causing mypy issues when using io.BytesIO

File: src/tentaclio/streams/base_stream.py [hunk 4]
```diff
@@ -98,7 +105,7 @@ class StreamerWriter(StreamBaseIO):
     connection atomic.
     """
 
-    def __init__(self, client: ContextManager[Streamer], buffer: IO):
+    def __init__(self, client: StreamerContextManager, buffer: IO):
```
Comment (j0nnyr0berts) [inline src/tentaclio/streams/base_stream.py:108]: Mypy doesn't seem to recognise the `ContextManager` will enter a `Streamer`, causing non `ContextManager` methods (e.g. `put`) to through type errors. Solved by creating protocol that inherits from both `ContextManager` and `Streamer`

File: Makefile (hunk 1 above)
Comment (capitancambio) [reply Makefile:21]: :roll_eyes:

File: src/tentaclio/fs/remover.py (hunk 2 above)
Comment (capitancambio) [reply src/tentaclio/fs/remover.py:11]: cool!

**Review by capitancambio — APPROVED** [review]
cool, I;'m happy to see those dummy context manager methods go with the the support to inherit from it

**Pull request merged**